pseudomonas_analysis/nbconverted/explore_expression_data.py

```diff
@@ -0,0 +1,634 @@
+)
+
+flat_simulated_expression = pd.read_csv(
+    simulated_expression_filename, 
+    sep="\t", 
+    header=0, 
+    index_col=0
+)
+
+flat_simulated_expression
+
+
+# In[12]:
+
+
+sns.distplot(flat_simulated_expression.loc["GSM822708_wtLB_A.CEL"])
+sns.distplot(flat_simulated_expression.loc["GSM822712_delta_cbrBLB_A.CEL"])
+
+
+# In[13]:
+
+
+# Heatmap visualization
+sns.heatmap(flat_simulated_expression.iloc[:,10:20])
+
+
+# In[14]:
+
+
+# Lets look at the DE stats associated with this flat volcano plot
+simulated_DE_filename = os.path.join(
+    local_dir,
+    "DE_stats",
+    f"DE_stats_simulated_data_{project_id}_5.txt"
+)
+
+flat_simulated_DE = pd.read_csv(
+    simulated_DE_filename, 
+    sep="\t", 
+    header=0, 
+    index_col=0
+)
+
+flat_simulated_DE
+
+
+# In[15]:
+
+
+# Lets look at the DE stats associated with DEGs volcano plot
+simulated_DE_filename = os.path.join(
+    local_dir,
+    "DE_stats",
+    f"DE_stats_simulated_data_{project_id}_11.txt"
+)
+
+simulated_DE = pd.read_csv(
+    simulated_DE_filename, 
+    sep="\t", 
+    header=0, 
+    index_col=0
+)
+
+simulated_DE
+
+
+# Flat volcano plots are due to samples between the two groups are very similar, so the log2 fold change is very small 10^-1 for each gene and so the adjusted p-values are nearly the same for each gene. Another artifact of the VAE variance shrinkage.
+
+# ## Explore leveling out of volcano plots
+# 
+# Why do volcano plots level out at at then extend outward? Is this an issue with the plotting?
+
+# In[16]:
+
+
+import matplotlib.pyplot as plt    
+def make_volcano_plot_simulated_notransform(
+    simulated_DE_stats_dir,
+    project_id,
+    pval_name,
+    logFC_name,
+    num_simulated,
+    ncols,
+    nrows,
+    fig_width,
+    fig_height,
+):
+    """
+	This function makes multiple volcano plots of example simulated experiments
+
+    Arguments
+    ----------
+    template_DE_stats_filename: str
+        File containing DE statistics for template experiment
+    project_id: str
+        Experiment identifier
+    pval_name: "padj" or "adj.P.Val"
+    logFC_name: "logFC" or "log2FoldChange"
+    num_simulated: int
+        Number of simulated experiments
+    ncols: int
+        Number of columns in facet plot
+    nrows: int
+        Number of rows in facet plot
+    fig_width: int
+        Width of figure
+    fig_height: ing
+        Height of figure
+
+	"""
+    fig, axes = plt.subplots(ncols=ncols, nrows=nrows, figsize=(fig_width, fig_height))
+
+    axes = axes.ravel()
+
+    for i in range(num_simulated):
+
+        # Get filename
+        simulated_DE_stats_filename = os.path.join(
+            simulated_DE_stats_dir, f"DE_stats_simulated_data_{project_id}_{i}.txt",
+        )
+
+        # Read simulated DE stats
+        simulated_DE_stats_df = pd.read_csv(
+            simulated_DE_stats_filename, sep="\t", index_col=0, header=0
+        )
+
+        # Take -log10 of adjusted p-value
```

Comment:
Looks like you're just taking the negative of the p-value here (not the -log10 like the comment suggests). Maybe you just copied over the code and forgot to update the comment?

Reply:
Good catch. I'll update this